Testing/VersionMinerTests/ActionInputTests.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+using CommandLine;
+using FluentAssertions;
+using VersionMiner;
+using VersionMinerTests.Helpers;
+
+// ReSharper disable UseObjectOrCollectionInitializer
+// ReSharper disable PossibleMultipleEnumeration
+namespace VersionMinerTests;
+
+/// <summary>
+/// Tests the <see cref="ActionInputs"/> class.
+/// </summary>
+public class ActionInputTests
+{
+    #region Prop Tests
+    [Fact]
+    public void Ctor_WhenConstructed_PropsHaveCorrectDefaultValuesAndDecoratedWithAttributes()
+    {
+        // Arrange & Act
+        var inputs = new ActionInputs();
+
+        // Assert
+        inputs.RepoOwner.Should().BeEmpty();
+        typeof(ActionInputs).GetProperty(nameof(ActionInputs.RepoOwner)).Should().BeDecoratedWith<OptionAttribute>();
+        inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.RepoOwner))
+            .AssertOptionAttrProps("repo-owner", true, "Gets or sets the owner of the repository.");
+
+        inputs.RepoName.Should().BeEmpty();
+        typeof(ActionInputs).GetProperty(nameof(ActionInputs.RepoName)).Should().BeDecoratedWith<OptionAttribute>();
+        inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.RepoName))
+            .AssertOptionAttrProps("repo-name", true, "Gets or sets the name of the repository.");
+
+        inputs.BranchName.Should().BeEmpty();
+        typeof(ActionInputs).GetProperty(nameof(ActionInputs.BranchName)).Should().BeDecoratedWith<OptionAttribute>();
+        inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.BranchName))
+            .AssertOptionAttrProps("branch-name", true, "Gets or sets the name of the branch where the file lives.");
+
+        inputs.FilePath.Should().BeEmpty();
+        typeof(ActionInputs).GetProperty(nameof(ActionInputs.FilePath)).Should().BeDecoratedWith<OptionAttribute>();
+        inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.FilePath))
+            .AssertOptionAttrProps("file-path", true, "Gets or sets the path to the file relative to the root of the repository.");
+
+        inputs.FileFormat.Should().BeEmpty();
+        typeof(ActionInputs).GetProperty(nameof(ActionInputs.FileFormat)).Should().BeDecoratedWith<OptionAttribute>();
+        inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.FileFormat))
+            .AssertOptionAttrProps("file-format", true, "The format of the data in the file that contains the version.  Currently the only supported format is 'xml' and is not case sensitive.");
+
+        inputs.VersionKeys.Should().NotBeNull();
+        inputs.VersionKeys.Should().BeEmpty();
+        typeof(ActionInputs).GetProperty(nameof(ActionInputs.VersionKeys)).Should().BeDecoratedWith<OptionAttribute>();
+        inputs.GetAttrFromProp<OptionAttribute>(nameof(ActionInputs.VersionKeys))
+            .AssertOptionAttrProps("version-keys", "The key(s) that can possibly hold the version in the file.");
```

Comment:
I don't know if this should be removed.

The key or keys might not contain a value.  The action can also be setup to NOT fail if the version is not found which means that it can be considered **ACCEPTABLE** to not have any versions.

Reply:
Per our conversation, I left "can" and removed "possibly"